pkg/deploy/controller/deployment/factory.go

```diff
@@ -122,6 +126,15 @@ func (factory *DeploymentControllerFactory) Create() controller.RunnableControll
 	}
 }
 
+// setPodAttributes updates the deployer pod attributes based on the deployment strategy configuration
+func (factory *DeploymentControllerFactory) setPodAttributes(strategy *deployapi.DeploymentStrategy, pod *kapi.Pod) {
+	switch strategy.Type {
+	case deployapi.DeploymentStrategyTypeCustom:
+		util.MergeInto(pod.Labels, strategy.Labels, 0)
```

Comment:
In case of conflict, we expect the strategy to override? Doesn't this make it possible to break the deployed pod label lookup the controller used to reap the pods?